nautobot_netbox_importer/diffsync/models/abstract.py

```diff
@@ -219,12 +219,19 @@ def create_nautobot_record(nautobot_model, ids: Mapping, attrs: Mapping, multiva
 
         return None
 
+    @classmethod
+    def special_clean(cls, diffsync, ids, attrs):  # pylint: disable=unused-argument
+        """Subclassable method called after `clean_ids` and `clean_attrs` to do any special cleaning needed."""
+
     @classmethod
     def create(cls, diffsync: DiffSync, ids: Mapping, attrs: Mapping) -> Optional["NautobotBaseModel"]:
         """Create an instance of this model, both in Nautobot and in DiffSync."""
         diffsync_ids, nautobot_ids = cls.clean_ids(diffsync, ids)
         diffsync_attrs, nautobot_attrs = cls.clean_attrs(diffsync, attrs)
 
+        cls.special_clean(diffsync, diffsync_ids, diffsync_attrs)
+        cls.special_clean(diffsync, nautobot_ids, nautobot_attrs)
```

Comment:
Adding another different clean method is a bit ugly but the set of attributes we need here is perpendicular to the set of attributes needed by `clean_ids`/`clean_attrs` so I can't use them for this specific purpose.